### ISSUE TITLE
pigpiod does not repsond to python test app

### ISSUE THREAD

**Issue by jonasdagefoerde** [joan2937/pigpio#2]
Hi Joan,
I installed the latest pigpio (via cloned repo) on my RPi with Raspbian linux and Python 3.2. After installation

```
make
make install
python3.2 setup.py install
```

I can successfully run (with all tests passing):

```
sudo ./x_pigpio
```

When I start the demon (sudo pigpiod) and try to open a socket in Python, I get a timeout (after some waiting):

```
>>> s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> s.connect(('localhost', 8888))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.error: [Errno 110] Connection timed out
```

Running pigpio.start() returns with the standard error message (and hints) after the timeout. I have no idea why this happens and pigpiod is not answering.
I ensured that the port 8888 is available before running pigpiod and that it is listened at after starting pigpiod.

Best
Jonas

**Comment by lucentbeam** [comment]
Hi, so I know this issue is closed, but I wanted to comment incase other encounter it. 

I was struggling with this quite a bit and only solved it after working with another program and realizing I couldn't ping localhost. It turns out when I had followed a guide to setting up an automatic wlan network (this is the culprit: http://weworkweplay.com/play/automatically-connect-a-raspberry-pi-to-a-wifi-network/), I had turned off the loopback network. So pigpiod would still run and listen on a port just fine, but the scripts could never open a socket and reach it. I believe this is why Jonas's issue was resolved by a reinstall.

In any case, to others who encounter this: make sure that there's a "LOOPBACK" network appearing when you type "ifconfig." If not, check that there's a line "auto lo" in your /etc/network/interfaces file. Here's an article that helped me: https://scottlinux.com/2012/06/09/cant-ping-localhost-in-linux/
